src/array.jl

```diff
@@ -26,11 +26,7 @@ for (A, V, M) in ((:CategoricalArray, :CategoricalVector, :CategoricalMatrix),
 
         $A{T, N}(::Type{T}, dims::NTuple{N,Int}; ordered=false) =
             $A(zeros(DefaultRefType, dims), CategoricalPool{T}(ordered))
-        function $A{T}(::Type{T}, dims::Int...; ordered=false)
-            A = $A{T}(dims)
-            ordered!(A, ordered)
-            A
-        end
+        $A{T}(::Type{T}, dims::Int...; ordered=false) = ordered!($A{T}(dims), ordered)
```

Comment:
It's a good sign that the change is good that we can simplify the implementation. I even wonder whether the `ordered` keyword argument is needed now.